packages/web3-eth/src/web3_eth.ts

```diff
@@ -308,7 +308,7 @@ export class Web3Eth extends Web3Context<Web3EthExecutionAPI, RegisteredSubscrip
 	 * ```
 	 */
 	public async calculateFeeData(
-		baseFeePerGasFactor = BigInt(2),
+		baseFeePerGasFactor: number = 2,
```

Comment:
This feature should be implemented with backward compatibility for any existing users. like 
`baseFeePerGasFactor: BigInt | number = BigInt(2),`